controllers/provider-aws/pkg/aws/secret.go

```diff
@@ -41,3 +46,19 @@ func ReadCredentialsSecret(secret *corev1.Secret) (*Credentials, error) {
 		SecretAccessKey: secretAccessKey,
 	}, nil
 }
+
+// NewClientFromSecretRef creates a new Client for the given AWS credentials from given k8s <secretRef> and
+// the AWS region <region>.
+func NewClientFromSecretRef(ctx context.Context, client client.Client, secretRef corev1.SecretReference, region string) (awsclient.Interface, error) {
```

Comment:
Can you make https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/pkg/controller/infrastructure/actuator_delete.go#L59-L67 calling this?

Reply:
Ahh, I did but there anyways it require accessing provider secret for further terraform related calls. So, left as it is. May be we need to refactor terraform related code. I'll prefer having separate PR for that.

Reply:
Ah, okay, got it. Makes sense, thanks.